modules/gamoshiBidAdapter.js

```diff
@@ -1,65 +1,73 @@
+    return url.substring(domainStart, url.indexOf('/', domainStart) < 0 ? url.length : url.indexOf('/', domainStart));
+  },
 
-function getTopWindowReferer() {
-  try {
-    return window.top.document.referrer;
-  } catch (e) {
-    utils.logMessage('Failed obtaining top window\'s referrer: ', e);
+  getTopWindowReferer: function () {
     try {
-      return window.document.referrer;
+      return window.top.document.referrer;
     } catch (e) {
-      utils.logMessage('Failed obtaining current window\'s referrer: ', e);
-      return '';
+      utils.logMessage('Failed obtaining top window\'s referrer: ', e);
+      try {
+        return window.document.referrer;
+      } catch (e) {
+        utils.logMessage('Failed obtaining current window\'s referrer: ', e);
+      }
     }
+    return '';
   }
 }
 
 export const spec = {
-  code: 'gambid',
-  aliases: [],
-  supportedMediaTypes: [ 'banner', 'video' ],
+  code: 'gamoshi',
+  aliases: ['gambid', 'cleanmedia'],
+  supportedMediaTypes: ['banner', 'video'],
 
-  isBidRequestValid: function(bid) {
+  isBidRequestValid: function (bid) {
     return !!bid.params.supplyPartnerId && typeof bid.params.supplyPartnerId === 'string' &&
-           (typeof bid.params[ 'rtbEndpoint' ] === 'undefined' || typeof bid.params[ 'rtbEndpoint' ] === 'string') &&
-           (typeof bid.params.bidfloor === 'undefined' || typeof bid.params.bidfloor === 'number') &&
-           (typeof bid.params[ 'adpos' ] === 'undefined' || typeof bid.params[ 'adpos' ] === 'number') &&
-           (typeof bid.params[ 'protocols' ] === 'undefined' || Array.isArray(bid.params[ 'protocols' ])) &&
-           (typeof bid.params.instl === 'undefined' || bid.params.instl === 0 || bid.params.instl === 1);
+      (typeof bid.params['rtbEndpoint'] === 'undefined' || typeof bid.params['rtbEndpoint'] === 'string') &&
+      (typeof bid.params.bidfloor === 'undefined' || typeof bid.params.bidfloor === 'number') &&
+      (typeof bid.params['adpos'] === 'undefined' || typeof bid.params['adpos'] === 'number') &&
+      (typeof bid.params['protocols'] === 'undefined' || Array.isArray(bid.params['protocols'])) &&
+      (typeof bid.params.instl === 'undefined' || bid.params.instl === 0 || bid.params.instl === 1);
   },
 
-  buildRequests: function(validBidRequests, bidderRequest) {
+  buildRequests: function (validBidRequests, bidderRequest) {
     return validBidRequests.map(bidRequest => {
-      const { adUnitCode, auctionId, mediaTypes, params, sizes, transactionId } = bidRequest;
-      const baseEndpoint = params[ 'rtbEndpoint' ] || 'https://rtb.gambid.io';
+      const {adUnitCode, auctionId, mediaTypes, params, sizes, transactionId} = bidRequest;
+      const baseEndpoint = params['rtbEndpoint'] || 'https://rtb.gamoshi.io';
       const rtbEndpoint = `${baseEndpoint}/r/${params.supplyPartnerId}/bidr?rformat=open_rtb&reqformat=rtb_json&bidder=prebid` + (params.query ? '&' + params.query : '');
+      let url = config.getConfig('pageUrl');
```

Comment:
`config.getConfig('pageUrl')` will never be used because if it exists it will trigger the `else` block below where it's reset to `utils.getTopWindowUrl()`
maybe:
`let url = config.getConfig('pageUrl') || bidderRequest && bidderRequest.refererInfo && bidderRequest.refererInfo.referer || utils.getTopWindowUrl();`

Reply:
which means there must not be a test verifying the expected behavior.

Reply:
@harpere fixed.

Reply:
@harpere please check it again now.

Reply:
@sa1omon - Sorry, logically that should work, but `lint` doesn't like it.  CircleCI error: `Unexpected mix of '||' and '&&'  no-mixed-operators`.